[CI] TransportGetEnrichPolicyActionTests.testGetMultiplePolicies failure

**Build scan**: https://gradle-enterprise.elastic.co/s/w6bw53hgqruvw

**Repro line**: `gradlew ':x-pack:plugin:enrich:test' --tests "org.elasticsearch.xpack.enrich.action.TransportGetEnrichPolicyActionTests.testGetMultiplePolicies" -Dtests.seed=289FE52574D88C9A -Dtests.security.manager=true -Dtests.locale=is-IS -Dtests.timezone=Africa/Lusaka -Druntime.java=8`

**Reproduces locally?**: Yes

**Applicable branches**: 7.9

**Failure history**: [Failed twice in the last 30 days](https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(_source),index:e58bf320-7efd-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'class:%22org.elasticsearch.xpack.enrich.action.TransportGetEnrichPolicyActionTests%22%20AND%20test:%22testGetMultiplePolicies%22'),sort:!(time,desc)))

**Failure excerpt**: 
```
  2> java.lang.AssertionError: 
    Expected: null
         but: was <ElasticsearchException[Enrich policy execution for [something-else] failed while validating field mappings for index [noed]]; nested: ElasticsearchException[Could not traverse mapping to field [gdtU]. Could not find the [gdtU] field under [root]];>
        at __randomizedtesting.SeedInfo.seed([289FE52574D88C9A:DC21962A5DEB0DE8]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.enrich.action.TransportGetEnrichPolicyActionTests.testGetMultiplePolicies(TransportGetEnrichPolicyActionTests.java:180)
```

--- a/x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/AbstractEnrichTestCase.java
+++ b/x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/AbstractEnrichTestCase.java
@@ -7,6 +7,7 @@ package org.elasticsearch.xpack.enrich;
 
 import org.elasticsearch.ResourceAlreadyExistsException;
 import org.elasticsearch.action.admin.indices.create.CreateIndexRequest;
+import org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest;
 import org.elasticsearch.client.Client;
 import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
 import org.elasticsearch.cluster.service.ClusterService;
@@ -66,7 +67,10 @@ public abstract class AbstractEnrichTestCase extends ESSingleNodeTestCase {
             try {
                 client.admin().indices().create(createIndexRequest).actionGet();
             } catch (ResourceAlreadyExistsException e) {
-                // and that is okay
+                // and that is okay, but update the mapping so that there is always a mapping for match field:
+                PutMappingRequest putMappingRequest = new PutMappingRequest(sourceIndex);
+                putMappingRequest.source(policy.getMatchField(), "type=keyword");
+                client.admin().indices().putMapping(putMappingRequest).actionGet();
             }
         }
     }

--- a/x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/action/TransportGetEnrichPolicyActionTests.java
+++ b/x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/action/TransportGetEnrichPolicyActionTests.java
@@ -175,7 +175,8 @@ public class TransportGetEnrichPolicyActionTests extends AbstractEnrichTestCase 
         assertThat(error.get(), nullValue());
 
         // save a second one to verify the count below on GET
-        error = saveEnrichPolicy("something-else", randomEnrichPolicy(XContentType.JSON), clusterService);
+        EnrichPolicy policy2 = randomEnrichPolicy(XContentType.JSON);
+        error = saveEnrichPolicy("something-else", policy2, clusterService);
         assertThat(error.get(), nullValue());
 
         final CountDownLatch latch = new CountDownLatch(1);

